src/plugins/discover/public/application/application.ts

```diff
@@ -12,28 +12,9 @@ import { renderApp as renderReactApp } from './index';
  * Here's where Discover is mounted and rendered
  */
 export async function renderApp(moduleName: string, element: HTMLElement) {
-  const app = mountDiscoverApp(moduleName, element);
+  // @ts-expect-error
```

Comment:
I know, not introduced by this PR, but this other method is inside a TS file. We should never use `@ts-expect-error` unless importing from JS files. We basically say here: Well you have types, and they say this does not work... so well just disable type checking for this (even worse than casting). Since you anyway touch the code, could we please fix the types of the `renderReactApp` method to not require this.

Reply:
sure thing!